mgmt-lambda/handlers/errorHandlers.ts

```diff
@@ -28,6 +30,21 @@ export async function handleWrongConfiguration(error: any): Promise<APIGatewayPr
   }
 }
 
+export function handleError(error: any): APIGatewayProxyResult {
+  if (error.name?.includes('AccessDenied')) {
+    error.code = ErrorCode.AWSAccessDenied
+  } else if (error.name === ResourceNotFoundException.name) {
+    error.code = ErrorCode.AWSResourceNotFound
+  }
+  return {
+    statusCode: 500,
+    body: JSON.stringify({ status: 'Error occurred', errorCode: error.code || ErrorCode.UnknownError }),
```

Comment:
the `error.code` can potentially leak information because `error` is not always our custom error object. We need to re-write this part.

Reply:
I think no one yet claimed `xErrorCode` before so using this instead of `code` 😄

Reply:
`xErrorCode` is confusing, and using a random field name is not the best solution here. Let's rollback this change, keep it `code` instead of `xErrorCode`, and work on this in a separate PR, what do you think? (What we lack is `try/catch` clause surrounding all code)